src/containers/SubscriberSignup/index.js

```diff
@@ -56,45 +70,80 @@ class SubscriberSignup extends React.Component {
         } = this.props;
         this.props.form.validateFields((err, values) => {
             if (!err) {
-              console.log('Received values of form: ', values);
               submitSubscriber(values);
             }
           });
     }
 
+    emailSearch(input) {
+        this.setState({
+            emailDataSource: this.props.allSubscribers.reduce((acc, curr) => {
+                if (curr.email.includes(input)) acc.push(curr.email);
```

Comment:
this sort of logic belongs in a selector. Generally you want components to receive the data in exactly the from they require.

Reply:
I hope it's alright that I put it in the reducer instead. I realize I could've put just the selected email address into app state, thereby making it available for a selector to do the filtering and return the correct subscriber. But since there has to be a reducer function anyway, it seemed simpler to me to have the logic in one place, make the subscriber-to-be-edited object part of state, and then the selector just returns that. Always open to feedback.